sklearn/model_selection/_search_successive_halving.py

```diff
@@ -66,6 +68,22 @@ class BaseSuccessiveHalving(BaseSearchCV):
     Zohar Karnin, Tomer Koren, Oren Somekh
     """
 
+    _parameter_constraints: dict = {
+        **BaseSearchCV._parameter_constraints,
+        # overwrite `scoring` since multi-metrics are not supported
+        "scoring": [StrOptions(set(get_scorer_names())), callable, None],
+        "random_state": ["random_state"],
+        "max_resources": [Integral, StrOptions({"auto"})],
+        "resource": [str],
```

Comment:
Here I was thinking to have the following:

```python
"resource": [str, StrOptions({"n_samples"})],
```

But apparently, we are going to generate a string as a `BadType` for some reason and doesn't raise the expected error. I assume passing `str` would be fine and caught by the more specific error later on.

Reply:
str seems good

> we are going to generate a string as a BadType

I don't think so. We generate an object of type ``BadType``, which works detecting that passing a non-string raises. And then we try to generate a string that is not in the options, but since all str are valid we don't raise the expected error